app/src/main/java/com/example/sharingang/models/Location.kt

```diff
@@ -0,0 +1,70 @@
+package com.example.sharingang.models
+
+import com.example.sharingang.utils.degreesToRadians
+import kotlin.math.atan2
+import kotlin.math.cos
+import kotlin.math.sin
+import kotlin.math.sqrt
+
+/**
+ * The radius of the earth, in meters
+ */
+private const val EARTH_RADIUS: Double = 6371e3
+
+/**
+ * Location represents a single point on the Earth.
+ *
+ * We use a latitude, and a longitude, both doubles. Now, doubles can be somewhat misleading,
+ * because the precision of the sensor data we receive is actually less accurate.
+ * On the other hand, it can't actually hurt to use more accurate numbers when doing
+ * calculations. We also don't perform many expensive calculations, so the use of Double
+ * isn't very onerous either.
+ */
+data class Location(val lat: Double, val lon: Double) {
+    /**
+     * This returns the distance along a great circle from this location to that location
+     *
+     * This is "as the crow flies", in the sense that the topography of the earth isn't
+     * taken into account, and the Earth is treated as a sphere.
+     *
+     * Unfortunately, in practice not only is the Earth not a sphere, but an elongated
+     * ellipsoid, because of its rotation, but there are also troublesome things like
+     * mountains, buildings, hills, etc.
+     *
+     * This distance is still a reasonable approximation.
+     *
+     * @param that the location to calculate the distance to, compared to this location
```

Comment:
The params would be "lat" and "long" instead of "that" I'd say.

Reply:
Well, there's only a single parameter `that` of type location, so this seems to be how javadoc should work, I think?

Reply:
Oh yeah my bad, I mixed the lat and long from the data class and thought it was in the function, whoops